### DISCUSSION TITLE
Next.js,  microfrontends and PWA

### DISCUSSION THREAD

**Discussion by enkicoma** [vercel/next.js#54624] [Help] [answered]
### Summary

Hi guys,

I want to understand from the architecture perspective if nextjs is good for microfrontends and PWA apps, especially for nextjs 13 with diff approach to routing and structure changes.

the main repo has been archived: https://github.com/shadowwalker/next-pwa

looks like nobody is supporting PWA here.

any links or repos that could help, will be much appreciated.

**Comment by e-roy** [comment]
Looks like someone forked `next-pwa` and is regularly maintaining a newer version.

[https://www.npmjs.com/package/@ducanh2912/next-pwa](https://www.npmjs.com/package/@ducanh2912/next-pwa)

[https://github.com/DuCanhGH/next-pwa](https://github.com/DuCanhGH/next-pwa)

I'm having decent luck so far in development with the `app` router.

Hope this helps anyone else searching

**Comment by leerob** [comment, marked as answer]
Hey everyone. We now have updated documentation on building PWAs with Next.js.

https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps
